defaults/main.yml

```diff
@@ -2,28 +2,51 @@
 # defaults file for ansible-role-proxmox-create-lxc/
 # vars needed to define the parameters for the wraped proxmox api
 
-node: my_node
-api_host: my_node.my_cluster.org
-api_user: deploy@pam
-node_deploy_password: D3pl0y_pwd
-url_ostemplate: http://download.proxmox.com/images/system/debian-10.0-standard_10.0-1_amd64.tar.gz
-# unprivileged: true    # Optional.
-cores: 1
-# cpu_limit: 1          # Optional.
-# cpu_units: 1000       # Optional.
-memory: 512
-# swap: 512             # Optional.
-disk: 32
-storage: local-lvm
-nameserver: 192.168.8.8 192.168.8.4
+pve_hostname: "{{ inventory_hostname.split('.')[0] }}"
+# By default, we suppose that `inventory_hostname` is the FQDN or the hostname of the host to create, so we set the variable to the hostname. 
+# You can arbitrarly define this hostname
 
+# node: my_node
```

Comment:
I understand the intention to give a transition period before removing backwards compatibility.

However, thinking about it a bit, the result of checking if the deprecated variables are defined, is a much more complex and "dirty" code, especially for those who come to the role in this version (v3.0.0). Isn't KISS better? On the other hand, it could cause an unwanted value to be taken by the role because someone forgot to update a definition with higher precedence.

In addition, a new major version is being released precisely here, so those who adopt it will already be aware that there are disruptive changes that will involve reviewing and updating the code of their own playbooks. Or else, they will remain using the previous version.

In the absence of a precise roadmap drawn up to release v4.0.0, my opinion is that it is more convenient to make the leap right here.

Reply:
Maybe you are right, and it's not a good idea to maintain API backwards compatibility. Facilitate (accelerate) code evolution versus complexities and superfluous patches which breake harmony of poetry... The question is valid, and I respect you prefer a cleaner code. 

I did that already for the [backupc_client role v2](https://github.com/UdelaRInterior/ansible-backuppc-client/blob/master/CHANGELOG.md), as a kind of geek exercice, and considering we had a lot of hosts using this role. So: how to merge the new version in the requirements.yml of master? With backwards compatibility, we can set the new version in master to start using it in new hosts, without going in the whole refactor, with each particular case, and letting each devop guy doing her job.

I don't think the code for this is so "complex and dirty": all is in `defaults/main.yml` and its quite clear (it even helps to update your API) and I don't see this risk: 
> it could cause an unwanted value to be taken by the role because someone forgot to update a definition with higher precedence.
Again: all is in `defaults/main.yml`, so as far as you define the new variable (which, with the new namespace, has much more chances to be unique) somewhere for the host, this value will be taken for the role, no matter what happens in defaults and/or the old API. 

Concerning your remark: 
> ... so those who adopt it will already be aware that there are disruptive changes that will involve reviewing and updating the code of their own playbooks.

Well, it's exactly what motivated this approach: I saw that many branches that involved heavier code refactor where delayed to be merged in master, breaking the: "release early, release often", and making the work harder while time goes by.

And I wouldn't do that for any role, only for these kind of roles that are deployed on many hosts, like `backuppc_client` or proxmox hosts creation. I didn't preserve the API backwards compatibility, for instance, for [backuppc server role](https://github.com/UdelaRInterior/ansible-backuppc), that was enhanced and refactored simultaneously: you usually have one, eventually a few backuppc server instances. So when you adopt the new major version of the role, you have to work it out. And there, you should start using the new major version and API of the backuppc client role. But then, you need a roadmap to deploy the new API in all your clients. The backwards compatibility of the API pretends to help in this deployment.

But well: in both cases, backuppc and proxmox_create, finally I updated the API of all the clients myself before merging... so it's not really useful (except to have two major version updates, which may make our roles to seem "mature" :-) .

Reply:
Concerning the roadmap, simple: the API is updated in our playbooks' set is, so as soon as we test this version, we can publish next major version, just dropping this backward's compatibility.

Reply:
For comment something: a default value is just a default value. For someone who recently arrived to the code it's confused.
I think that ansible is not a programming language, is a way to describe our infrastructure as code.
These changes is not KISS...

Reply:
>  how to merge the new version in the requirements.yml of master?

Merging this new version in our particular IaC project will be the same in either of the two cases, the refactor will be done anyway, we just have to decide whether to do it all at once or extend it over time by making new hosts coexist in the recent structure with old hosts that may or may not have been upgraded by a devop guy.

For branches that have delayed behind *master*, its requirements.yml will continue pointing to the v2.x.y of the role so consistency and integrity will be maintained.

> I don't think the code for this is so "complex and dirty"

I consider that it is for those who wish to use the role from this new version and have no idea or interest in the structure and names of previous variables.

> and I don't see this risk

You are right! After reviewing it again, the risk of taking unwanted values is practically nil because it is precisely in `defaults/main.yml` as you say.

In conclusion, both ways are equally valid. It's a matter of choosing the one that best suits.

I give my vote to KISS

Reply:
Let's do it simple, without re-writing history: in our IaC, we no longer need this backward's compatibility, if ever we did. So let's prepare and publish the next major version, dropping this questioned code. Do you both agree?

I started [for proxmox_create_lxc](https://github.com/UdelaRInterior/ansible-role-proxmox-create-lxc/pull/12/commits/09e58a91d5c918d669519c5b8992f6cab9d9bb26), tested with our host `pavada`.

Reply:
And I did it also [for KVM role](https://github.com/UdelaRInterior/ansible-role-proxmox-create-kvm/pull/9). 

Both basically tested with an LXC and a KVM creation.

Reply:
> tested with our host `pavada`.

Don't you think that is a "`pavada`", don't you? :-)

Reply:
In French we would say: "Les doigts dans le nez!" (_Finger in the nose!_), that could be translated by: "Una pavda!" ...and in the US you could do that for Thanksgiving.